streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -607,8 +614,27 @@ class StreamingContext private[streaming] (
  */
 
 object StreamingContext extends Logging {
+  /**
+   * Lock that guards access to global variables that track active StreamingContext.
+   */
+  private val ACTIVATION_LOCK = new Object()
 
-  private[streaming] val DEFAULT_CLEANER_TTL = 3600
+  private var activeContext: AtomicReference[StreamingContext] =
+    new AtomicReference[StreamingContext](null)
+
+  private def assertNoOtherContextIsActive(): Unit = {
+    ACTIVATION_LOCK.synchronized {
+      if (activeContext.get() != null) {
+        throw new SparkException("Only one StreamingContext may be started in this JVM")
```

Comment:
Well, let me be a little bit more unambiguous in the choice of verbs. We are allowed to instantiate multiple StreamingContexts provided only one is started (`ssc.start()`) at a time. That's why I used "started" so that it directly maps to `ssc.start()`. This is different from `SparkContexts` where it is started as soon as it is instantiated.

I will add the callsite information.